rfcs/0000-light-rail-trip-model.md

```diff
@@ -0,0 +1,225 @@
+
+This intermediate state is proposed to allow for a more incremental implementation that's easier to deliver and provides some value without requiring too much deep refactoring of RTR. It can optionally be skipped if the Glides and Transit Data teams collectively decide that it creates more net work for not enough additional benefit.
+
+## Train - trip assignment feed
+
+In a future state, Glides will be responsible for light rail trip assignments, and generate a new event feed of train-to-trip assignment events. RTR will use this as its primary source of information 
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## GTFS and TODS schedule data
+
+Light rail schedule data in GTFS and (eventually) TODS files will be produced by `gtfs_creator` based on an export from HASTUS. The HASTUS trips for the lead car will be the basis for GTFS trips, and GTFS trips will take on their IDs. Integrating the schedule for multi-car trips (that is, trips with a trailer) into the TODS data will be defined in a separate discussion about the adoption of TODS. When `gtfs_creator` takes a trip from HASTUS and outputs it with only minor factual corrections, it SHOULD retain the original trip ID. However, if a trip is actively modified as part of the `gtfs_creator` disruption modeling process, its ID MUST be changed so as to not overlap with the HASTUS trip IDs.
+
+## Glides schedule import
+
+Glides will import schedule data from either its own HASTUS export or, in the future, TODS. In either case, trip data is loaded into the `scheduled_trips` table. The HASTUS trip ID will be used directly to populate the `trip_id` column. While working directly with HASTUS, some modifications will be necessary to which IDs we actually use. Currently, our export gets a column called `trp_number`, but the actual, database-unique ID is called `trp_int_number`, and the latter is also the value published by `gtfs_creator`. The significant different here is that cars grouped into one logical trip all have the same `trp_number` generally, but as `trp_int_number` is database-unique, that value will differ. We want the benefits of both combined trip information for multiple cars as well as uniqueness and agreement between systems. To achieve this, we will combine the trips for the different constituent cars and determine the ultimate ID in the following manner:
+
+1. Group the trip entries for each schedule ID by the `trp_number` value.
```

Comment:
We've noticed issues with using `trp_number` in the past: There was that one time when some trips were added to the rating at the last minute, and they ended up all having the same `trp_number`. I also think that if a two-car pullout trip has different times (because the pilot has to show up a couple minutes earlier than the trailer), they could end up with different `trp_number`s, even when we want to consider them the same trip (I haven't double checked this).

So, I think that `trp_number` may not be reliable for this purpose. Perhaps the RFC should not say _how_ the cars are grouped together into a single trip, and instead leave that up to the implementer, so they could instead base the groups on time+location if it's more reliable.

This comment does not affect 2. and 3., which are only about what to do with the groups, it doesn't matter how they're formed.

Reply:
I seemed to remember something like that happening in the past but then was having a very hard time digging up an actual example when working on this RFC, which makes me wonder if I actually remembered correctly. I also don't want to overcorrect for something that happened once. I made note of it in 88e2688 though.